google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -237,6 +237,10 @@
   }
 
   private final BigQueryRpc bigQueryRpc;
+  private static final BigQueryRetryConfig DEFAULT_RATE_LIMIT_EXCEEDED_RETRY_CONFIG =
+      BigQueryRetryConfig.newBuilder()
+          .retryOnMessage(BigQueryErrorMessages.RATE_LIMIT_EXCEEDED_MSG)
```

Comment:
I find this a little extraneous -- why do we need to specify every message we need to retry on? If we build a config, I imagine we just retry on all the specified error messages in the config.

Reply:
Sure Stephanie. As discussed, I have tried to create it on the lines of `com.google.cloud.ExceptionHandler` and this may give us additional flexibility to configure hooks like `retryOnStatus` or `retryOnReason` later as necessary. Happy to refactor it as needed.